skbio/tree/_tree.py

```diff
@@ -617,6 +617,75 @@ def lowest_common_ancestor(self, tipnames):
 
     lca = lowest_common_ancestor  # for convenience
 
+    def path(self, other, include_ends=False):
+        r"""Return the list of nodes in the path from one node to another.
+
+        Parameters
+        ----------
+        other : TreeNode
+            Final node of path.
+        include_ends: bool, optional
+            Whether to include the initial and final nodes in the list.
+
+        Returns
+        -------
+        list
+            List of TreeNode objects.
+
+        Examples
+        --------
+        >>> from skbio import TreeNode
+        >>> tree = TreeNode.read(["((a,b)c,(d,e)f)root;"])
+        >>> node_1, node_2 = tree.find('a'), tree.find('d')
+        >>> path = node_1.path(node_2)
+        >>> print(len(path))
+        3
+        >>> path_2 = node_1.path(node_2, include_ends=True)
+        >>> print(len(path_2))
+        5
+
+        """
+        # create list of ancestors including nodes themselves
+        anc1, anc2 = (
+            [self] + self.ancestors(),
+            [other] + other.ancestors(),
+        )
+
+        # initialize variables
+        lca = None
+        lca_i = None
+        path = None
+
+        # find lowest common ancestor
+        for i, (n1, n2) in enumerate(zip(reversed(anc1), reversed(anc2))):
+            if n1 is n2:
+                lca = n1
+                lca_i = i
+            else:
+                break
+
+        # check to see if nodes are on same tree
+        if lca is None:
+            raise TypeError("Could not find path between nodes.")
+
+        # handle cases for nodes that are ancestors of each other
+        if self in anc2:
+            path = anc2[: len(anc2) - lca_i][::-1]
+        elif other in anc1:
```

Comment:
These `other in anc1` can be replaced with `self is lca`, which is faster. But as I guessed above, you probably don't need this check.